### ISSUE TITLE
RNIap.requestPurchase have no resolved promise result on iOS 14.0 version

### ISSUE THREAD

**Issue by VEnom42** [dooboolab/react-native-iap#1058]
### Version of react-native-iap
4.4.8, 4.5.0
### Version of react-native
0.61.5
### Platforms you faced the error (IOS or Android or both?)
IOS (14.0)
### Expected behavior
requestSubscription and requestPurchase methods should resolve promise on sucessful purchase.
### Actual behavior
`purchaseUpdatedListener` receives the result of successful purchase but `RNIap.requestPurchase(sku, false)` and `RNIap.requestSubscription(sku, false)` have no resolved promise result. Actually there is no success callback execution. It works fine in previous ios versions and in android.
### Tested environment (Emulator? Real Device?)
both
### Steps to reproduce the behavior
1. Set up purchases.
2. Get real device with ios 14.0 or emulator with working [StoreKit](https://developer.apple.com/documentation/storekit/in-app_purchase/testing_in-app_purchases_in_xcode)
3. Try to get console.log from successful purchase:
```ts
RNIap.requestPurchase(sku, false).then(result => {console.log('result', result)});
//or
RNIap.requestSubscription(sku, false).then(result => {console.log('result', result)});
```

**Comment by MacroBet** [comment]
I'm having the same problem

**Comment by captaincole** [comment]
Also seeing this issue on iOS 14 beta 2.

**Comment by corasan** [comment]
Weird thing is that it was working for me perfectly fine.

**Comment by lempa** [comment]
We're having the same issue  in a mac catalyst app on iOS 13 / macOS 10.15.5.
The steps to reproduce the behavior are the same

**Comment by iaphub** [comment]
Just submitted a pull request to fix the issue.
As soon as it is accepted, we will update [react-native-iaphub](https://github.com/iaphub/react-native-iaphub) to the latest version of react-native-iap with the fix 👍

**Comment by VEnom42** [comment]
Fixed in `4.5.2`.

**Comment by n0umankhan** [comment]
+1 still having the same problem in IOS, I'm using 4.5.3
